code/modules/mob/living/carbon/carbon.dm

```diff
@@ -514,11 +524,14 @@ GLOBAL_LIST_INIT(ventcrawl_machinery, list(/obj/machinery/atmospherics/unary/ven
 
 /mob/living/update_pipe_vision()
 	if(pipes_shown.len)
-		if(!is_ventcrawling(src))
-			remove_ventcrawl()
+		remove_ventcrawl()
+		if(is_ventcrawling(src))
+			add_ventcrawl(loc)
```

Comment:
Won't this add duplicates to the `ventcrawler` global since this is also called when a pipe gets destroyed/created?

Reply:
It shouldn't as this check will do the following.
First, check if the mob currently has a pipe overlay, if they do. Delete it. Then if they are still currently in a pipe, regenerate it.
Then if they don't have an overlay, check if they are in a pipe, if they are, give them an overlay. Then just to be sure, if they are not in a pipe at this point, delete an overlay if they have one (This is just a sanity check to cover all four cases and probably unneeded)

The four cases are as follows for a nested if

They have an overlay and are in a pipe. Remove and readd.
They have an overlay and are not in a pipe. Just remove.
They don't have an overlay and are in a pipe. Give them an overlay.
They don't have an overlay and are not in a pipe. Try to remove it just to be sure.

Reply:
This would be slightly more expensive, but would always work. Thoughts on this as a replacement?

/mob/living/update_pipe_vision()
	remove_ventcrawl()
	if(is_ventcrawling(src))
		add_ventcrawl(loc)

Reply:
All gut here. Overlooked some code it seems back when I commented on it.